docs/content/en/docs/reference/clusterspec/cloudstack.md

```diff
@@ -5,4 +5,346 @@ weight: 30
+You may define one or more worker node groups.
+
+### workerNodeGroupConfigurations.count (required)
+Number of worker nodes
+
+### workerNodeGroupConfigurations.machineGroupRef (required)
+Refers to the Kubernetes object with CloudStack specific configuration for your nodes. See `CloudStackMachineConfig Fields` below.
+
+### workerNodeGroupConfigurations.name (required)
+Name of the worker node group (default: md-0)
+
+### workerNodeGroupConfigurations.taints
+A list of taints to apply to the nodes in the worker node group.
+
+Modifying the taints associated with a worker node group configuration will cause new nodes to be rolled-out, replacing the existing nodes associated with the configuration.
+
+At least one node group must not have `NoSchedule` or `NoExecute` taints applied to it.
+
+### workerNodeGroupConfigurations.labels
+A list of labels to apply to the nodes in the worker node group. This is in addition to the labels that
+EKS Anywhere will add by default.
+For example, this label identifies a Cluster API failuredomain, which causes CAPC to substitute the value for the actual AZ name as a label on the nodes:
+
+```yaml
+    labels:
+      cluster.x-k8s.io/failure-domain: ds.meta_data.failuredomain
+```
+
+Modifying the labels associated with a worker node group configuration will cause new nodes to be rolled out, replacing
+the existing nodes associated with the configuration.
+
+## CloudStackDatacenterConfig
+
+### availabilityZones.account (optional)
+Account used to access CloudStack.
+The default is `admin`.
+As long as you pass valid credentials, through `availabilityZones.credentialsRef`, this value is not required.
+
+### availabilityZones.credentialsRef (optional)
+If you passed credentials through the environment variable `EKSA_CLOUDSTACK_B64ENCODED_SECRET` noted in [Create CloudStack production cluster]({{< relref "../../getting-started/production-environment/cloudstack.md" >}}), you can identify those credentials here.
+For that example, you would use the value `global`.
+You can also use a previously created Kubernetes secret.
+
+### availabilityZones.domain (required)
+CloudStack domain to deploy the cluster. The default is `ROOT`.
+
+### availabilityZones.managementApiEndpoint (required)
+Location of the CloudStack API management endpoint. For example, `http://10.11.0.2:8080/client/api`.
+
+### availabilityZones.name (required)
+Name of the CloudStack zone on which to deploy the cluster.
+As an alternative, you could pass a Zone ID to `availabilityZones.id`.
+
+### availabilityZones.zone.network.name (required)
+CloudStack network name to use with the cluster.
+
+## CloudStackMachineConfig
+In the example above, there are separate `CloudStackMachineConfig` sections for the control plane (`my-cluster-name-cp`), worker (`my-cluster-name`) and etcd (`my-cluster-name-etcd`) nodes.
+
+### computeOfferings (required)
+Name or ID of the CloudStack compute instance.
+
+### users[0].name (optional)
```

Comment:
I think you can pass in an arbitrary number of user objects, not just 1

Reply:
Done.